components/relay/src/error.rs

```diff
@@ -4,23 +4,41 @@
 
 pub type Result<T> = std::result::Result<T, Error>;
 
-#[derive(Debug, thiserror::Error)]
+#[derive(Debug, thiserror::Error, uniffi::Error)]
 pub enum Error {
     #[error("Network error: {0}")]
-    Network(#[from] viaduct::Error),
+    Network(String),
 
     #[error("UTF-8 error: {0}")]
-    Utf8(#[from] std::string::FromUtf8Error),
+    Utf8(String),
 
     #[error("URL parse error: {0}")]
-    UrlParse(#[from] url::ParseError),
+    UrlParse(String),
 
     #[error("JSON error: {0}")]
-    Json(#[from] serde_json::Error),
+    Json(String),
 
     #[error("Status error: {0}")]
-    Status(#[from] viaduct::UnexpectedStatus),
+    Status(String),
 
     #[error("Relay API returned an error: {0}")]
     RelayApi(String),
 }
+
+impl From<url::ParseError> for Error {
+    fn from(err: url::ParseError) -> Self {
+        Error::RelayApi(format!("URL parse error: {}", err))
+    }
+}
+
+impl From<viaduct::Error> for Error {
+    fn from(err: viaduct::Error) -> Self {
+        Error::RelayApi(format!("Viaduct error: {}", err))
+    }
+}
+
+impl From<serde_json::Error> for Error {
+    fn from(err: serde_json::Error) -> Self {
+        Error::RelayApi(format!("JSON error: {}", err))
+    }
+}
```

Comment:
I'd recommend keeping the old error code, which uses separate variants for each of these errors.  It's often useful to distinguish the cases when you're handling them internally.

OTOH, consumers of your component might not care about these details.  In that case, you can define a second error enum, for the public API along with logic to convert between them.  The other thing we typically do is report errors to sentry during the internal -> public error conversion, if we see an internal error that we don't expect.  Check out [errors.rs](https://github.com/mozilla/application-services/blob/main/components/example/src/error.rs) from the example crate for details.

Reply:
I added https://github.com/mozilla/application-services/pull/6773/commits/8a2d05d098ab252a4fda1ad2b5b8b766d0cde573 defining the second error enum for the public API and logic to convert between them using the code from the example errors. Check that I did it alright?